Add optional low_background_flag to df_f_calculation.py to curb unrealistically high dF/F values

liorgolgher opened the pull request:
Added an optional boolean low_background_flag: False (default) if the baseline is strictly positive, as in analog imaging. True for low background recordings, e.g. photon counting, three-photon excitation or sparse acquisition.
If True, delta_F is divided by (f0 + low_background_offset_value), artificially increasing the baseline for low-background imaging modalities, so as to prevent unrealistically high dF/F values.

liorgolgher committed:
Update df_f_calculation.py

Added an optional boolean low_background_flag: False (default) if the baseline is strictly positive, as in analog imaging. True for low background recordings, e.g. photon counting, three-photon excitation or sparse acquisition.
If True, delta_F is divided by (f0 + low_background_offset_value), artificially increasing the baseline for low-background imaging modalities, so as to prevent unrealistically high dF/F values.

liorgolgher committed:
minor change: input data is fluorescence traces

HagaiHargil commented on dff_calc/df_f_calculation.py:
```diff
@@ -9,6 +9,7 @@ def dff_calc(
     tau_1: float = 0.35,
     tau_2: float = 2.0,
     invert: bool = False,
+    low_background_flag: bool = False,
```
Ditch the `flag` suffix :)

HagaiHargil replied:
Perhaps its better if the argument is the actual offset value, and when it's 0 nothing will happen? We can have a default value of 0 so that it's off by default.

HagaiHargil commented on dff_calc/df_f_calculation.py:
```diff
@@ -35,20 +36,23 @@ def dff_calc(
         False (default) if the transient is expected to be positive, True
         otherwise.
+    low_background_flag : bool, optional
+        False (default) if the baseline is strictly positive, as in analog imaging. True for low background recordings, e.g. photon counting, three-photon excitation or sparse acquisition.
```
Change the `flag` suffix here as well, and wrap lines at about 80 columns please.

HagaiHargil reviewed (changes requested):
Thanks!

As I said, I think we should have the actual offset value acting as the flag as well, with a default value of 0.

liorgolgher committed:
Adding 2*low_background_offset_value to F-F0

Adding 2*low_background_offset_value to the numerator F - F0, to prevent dF/F negative values when F0 is nonzero while F is zero.
